### PR TITLE
RoutineDetailView first PR

### PR DESCRIPTION
**Pull request by YLee2121**
# RoutineDetailView first PR

## 🧾Summary

* `RoutineDetailView()` with two subviews `RDetailView_ESummaryRow()`, `RDetailView_SetRow()`.

* Mock Model: `RoutineModel`, `ExerciseModel`, and `ExerciseSetModel`.

* Mock ViewModel: `RDetailViewModel`

* All the mock Models and mock ViewModel need to be refined in the future. ❗❗❗

## 🕵️‍♂️PR Detail  

* `RoutineDetailView()`

  * Routine summary data on top of the view as a title.
  
  * Each exercise is a `Section`, and each exercise set is a list row within the `Section`.

  * `Section` header has a button to go to the `Chart` module. The button will go to the `Chart` module. You can see this function from `MainView.swift` because `TabView` is **only** constructed in `MainView.swift`.

  * The first list row of exercise `Section` is a summary statistics of the exercise.

## 📸Screenshots

imageimage

### CODE REVIEW

File: GymBro/Views/HistoryViews/RoutineDetailView.swift [hunk 1]
```diff
@@ -7,14 +7,156 @@
...
+/// Detail for each routine
+///
+/// The view comes after `IntroHistoryView`
 struct RoutineDetailView: View {
+    
+    
+    // MARK: PROPERTY
+    
+    /// Mock ViewModel Singleton
+    @ObservedObject var vm = RDetailViewModel.instance
```
Comment (alex308248) [inline GymBro/Views/HistoryViews/RoutineDetailView.swift]: It;s better to change to a more meaningful name. Somethin like detailViewModel, etc.

Comment (YLee2121) [reply GymBro/Views/HistoryViews/RoutineDetailView.swift]: Done rename!

**Review by alex308248 — APPROVED** [review]
Nice PR. Need to pull the latest version of version1 and merge!

**Comment by YLee2121** [comment]
> Nice PR. Need to pull the latest version of version1 and merge!

Will fix!

**Pull request merged**